File: app/src/main/java/com/pnuema/bible/retrievers/BaseRetriever.java
package com.pnuema.bible.retrievers;

import java.util.Observable;

public abstract class BaseRetriever extends Observable {
    BaseRetriever() {
    }

    String getTag() {
        return getClass().getSimpleName();
    }

    public abstract void savePrefs();

    public abstract void readPrefs();

    public abstract void getVersions();

    public abstract void getBooks();

    public abstract void getChapters(String book);

    public abstract void getVerseCount(String version, String book, String chapter);

    public abstract void getVerses(String version, String book, String chapter);
}

File: app/src/main/java/com/pnuema/android/bible/retrievers/BaseRetriever.kt
package com.pnuema.android.bible.retrievers

import java.util.*

abstract class BaseRetriever internal constructor() : Observable() {

    val tag: String
        get() = javaClass.simpleName

    abstract fun savePrefs()

    abstract fun readPrefs()

    abstract fun getVersions()

    abstract fun getBooks()

    abstract fun getChapters(book: String)

    abstract fun getVerseCount(version: String, book: String, chapter: String)

    abstract fun getVerses(version: String, book: String, chapter: String)
}
